src/hooks/useSelectedRevisions.ts

```diff
@@ -21,9 +20,12 @@ const useSelectRevision = () => {
   const addSelectedRevisions = () => {
     const newSelected = [...selectedRevisions];
     newSelected.push(...baseCheckedRevisions, ...newCheckedRevisions);
+    const filteredSelected = newSelected.filter(
+      (revision, index, self) =>
+        self.findIndex((r) => r.id === revision.id) === index,
+    );
```

Comment:
Is it still needed now that you removed the duplicates?
If yes, then I don't understand really what this is doing :-)

Reply:
I'll add a comment. This is to remove duplicates in the revisions not the repositories. For example, if a user selects the same revisions for base and new.